internal/resources/slo/resource_slo.go

```diff
@@ -57,17 +57,44 @@ Resource manages Grafana SLOs.
 						"freeform": &schema.Schema{
 							Type:     schema.TypeList,
 							MaxItems: 1,
-							Required: true,
+							Optional: true,
 							Elem: &schema.Resource{
 								Schema: map[string]*schema.Schema{
 									"query": &schema.Schema{
 										Type:        schema.TypeString,
-										Optional:    true,
+										Required:    true,
 										Description: "Freeform Query Field",
 									},
 								},
 							},
 						},
+						"ratio": &schema.Schema{
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"success_metric": &schema.Schema{
+										Type:        schema.TypeString,
+										Description: `Metric for success events (numerator)`,
+										Required:    true,
+									},
+									"total_metric": &schema.Schema{
+										Type:        schema.TypeString,
+										Description: `Metric for total events (denominator)`,
+										Required:    true,
+									},
+									"group_by_labels": &schema.Schema{
+										Type:        schema.TypeList,
+										Description: `Defines Group By Labels used for per-label alerting. Also appear as variables on SLO dashboards to enable filtering and aggregation `,
```

Comment:
if this is the place end users will find documenation, then we should mention here that GroupByLabels are optional, but need to confirm to prometheus label name syntax 

https://github.com/grafana/slo/blob/558e3c81683d907b5f36accc81eaed086442d3d2/schemas/slo_api.cue#L135